### ISSUE TITLE
Undisimssable Progress Dialog in StockMediaPickerActivity

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/WordPress-Android#7628]
### Expected behavior
When trying to set a blog's featured image from the StockMediaPicker, I want to be able to rotate the device and still have things work seamlessly, adding then picture and going back to the Post Settings screen without harm.

### Actual behavior
When trying to set a blog's featured image from the StockMediaPicker, if you rotate the device several times there's a high chance the progress dialog will be stuck there forever, and cannot be dismissed. You have to kill the app to start again.

![progressdismiss](https://user-images.githubusercontent.com/6597771/38575274-6f3516f2-3cd1-11e8-86d6-a57fcd7d1d8b.gif)

### Steps to reproduce the behavior
1. start a draft post
2. optionally enter a title and some content
3. open the overflow menu (tap on the three dots at the upper right corner of the screen, on the action bar)
4. tap on "Post Settings"
5. scroll down to the section labelled "Featured Image"
6. tap on the `SET FEATURED IMAGE` button
7. when the image picker appears, tap on the first icon to the left, and then choose `Choose from Free Photo Library`
8. enter a search term, for example "green", and wait for the resulting images to appear
9. pick one image, then tap on the bottom button `USE THIS PHOTO`.
10. as the "Uploading media..." progress dialog appears, rotate the device many times (4 or 5 times, or for arpproximately 6 to 10 seconds). 
11. observe the progress dialog never goes away, and you can't even tap the `BACK` button.

### Technical notes
Setting the device’s developer mode ON, and then using the “Don’t keep activities” option, try to select a stock image and as soon as you tap `USE THIS PHOTO` start rotating the device several times. You’ll end up with an undismissable “Uploading media…”  dialog, because in the creating/destroying of the Activity the FluxC event for `MediaStore.OnStockMediaUploaded` never arrives (the dispatcher is unregistered, the event occurs, and then we re-register, thus missing it).

As discussed on Slack, this is a problem that: 

- a) is probably happening in many other places of our app, and 
- b) is a candidate situation to be solved with the `ViewModel` pattern

cc @nbradbury @oguzkocer 

##### Tested on [Moto X], Android [5.1], and Google Pixel,  Android 8.0

**Comment by designsimply** [comment]
I tested with Pixel 3, Android 9, WPAndroid 11.8 alpha-152 using the steps provided and was unable to reproduce the problem. I tested with 3 types of sites: WP.com with plugins, WP.com simple, Jetpack 7.0.1 (this one is shown in the video).  ([38s](https://cloudup.com/cbwzUV8bpvc))

@mzorz should this issue remain open because it still might be happening on Android 8? Should it remain `[Pri] High` even though it hasn't had activity in 10+ months?
